src/common/utils/utils.ts

```diff
@@ -343,3 +343,18 @@ export function removeTrailingSlash(url?: string) {
   if (!url) return '';
   return url.replace(/\/$/, '');
 }
+
+export function promiseWrapper<T, Args extends any[]>(
```

Comment:
Can you explain the purpose of this function?

Reply:
I think this PR was accidentally branched from this [PR](https://github.com/hirosystems/explorer/pull/1687), so it has all those changes. I made a new PR that removed all those changes. The mobile menu PR is a 1 liner. The new PR is [here](https://github.com/hirosystems/explorer/compare/fix/mobile-menu2?expand=1).

But to ansewr the question, this is for my fix for the delete network bug. I saw that you had made addCustomNetwork return a promise, just so you could add a .then hook to it. Instead of muddying the implementation of addCustomNetwork I added this utility function that allows you to take a function, wrap it in a promise that resolves after the function is called, which allows us to add .then hooks to any function we want